cdk-infra/lib/constructs/worker/buckets-construct.ts

```diff
@@ -1,159 +1,29 @@
-            hostName,
-            replaceKey: {
-              prefixWithKey: `${prefixUrl}/atlas/cli/v1.1`,
-            },
-          },
-          {
-            condition: {
-              keyPrefixEquals: `${prefixUrl}/atlas/cli/v1.1.5`,
-            },
-            protocol: RedirectProtocol.HTTPS,
-            hostName,
-            replaceKey: {
-              prefixWithKey: `${prefixUrl}/atlas/cli/v1.1`,
-            },
-          },
-          {
-            condition: {
-              keyPrefixEquals: `${prefixUrl}/atlas/cli/v1.1.4`,
-            },
-            protocol: RedirectProtocol.HTTPS,
-            hostName,
-            replaceKey: {
-              prefixWithKey: `${prefixUrl}/atlas/cli/v1.1`,
-            },
-          },
-          {
-            condition: {
-              keyPrefixEquals: `${prefixUrl}/atlas/cli/v1.1.3`,
-            },
-            protocol: RedirectProtocol.HTTPS,
-            hostName,
-            replaceKey: {
-              prefixWithKey: `${prefixUrl}/atlas/cli/v1.1`,
-            },
-          },
-          {
-            condition: {
-              keyPrefixEquals: `${prefixUrl}/atlas/cli/v1.1.2`,
-            },
-            protocol: RedirectProtocol.HTTPS,
-            hostName,
-            replaceKey: {
-              prefixWithKey: `${prefixUrl}/atlas/cli/v1.1`,
-            },
-          },
-          {
-            condition: {
-              keyPrefixEquals: `${prefixUrl}/atlas/cli/v1.1.0`,
-            },
-            protocol: RedirectProtocol.HTTPS,
-            hostName,
-            replaceKey: {
-              prefixWithKey: `${prefixUrl}/atlas/cli/v1.1`,
-            },
-          },
-        ];
-      }
-
+    const buckets: IBucket[] = docsBucketNames.map((bucketName) => {
       const featureName = getFeatureName();
+      const useCustomBuckets = getUseCustomBuckets();
+
+      // If we want to use buckets that don't currently exist, we can call this method to create
+      // them for individual testing purposes
+      if (useCustomBuckets) return createCustomBucket({ scope: this, featureName, env, bucketName });
 
-      const stackBucketName = `${featureName}-${bucketName}-${env}`.toLowerCase();
+      const bucketEnv = env === 'prd' ? 'prd-staging' : env;
+      const stackBucketName = `${featureName}-${bucketName}-${bucketEnv}`.toLowerCase();
 
-      const bucket = new Bucket(this, stackBucketName, {
-        removalPolicy: RemovalPolicy.DESTROY,
-        websiteRoutingRules,
-        bucketName: stackBucketName,
-        websiteIndexDocument: 'index.html',
-        websiteErrorDocument: 'docs-qa/404/index.html',
-        blockPublicAccess: new BlockPublicAccess({
-          blockPublicAcls: false,
-          blockPublicPolicy: false,
-          ignorePublicAcls: false,
-          restrictPublicBuckets: false,
-        }),
-      });
+      const bucket = Bucket.fromBucketName(this, stackBucketName, `${bucketName}-${bucketEnv}`);
```

Comment:
The `fromBucketName` method requires a unique ID value to identify the construct, which is the second argument of the method. I believe this is just necessary for CDK to uniquely identify constructs that are instantiated 🤪. I arbitrarily chose `stackBucketName` to be a combination of `${featureName}-${bucketName}-${bucketEnv}`, but it's also fine if I just did the bucket-name with the env. 

We exclude the `featureName` for the final argument because that is the name of the bucket we want to retrieve.

There won't be a conflict if 2 different features referred to the same bucket as we are retrieving that bucket so that we can grant permission for the enhanced autobuilder to read and write to the bucket. We actually do this now, these buckets are currently used by the standard autobuilder and the enhanced currently.

Reply:
Oh, gotcha! For some reason I thought we were creating a new set of isolated `${featureName}-${bucketName}-${bucketEnv}` buckets per feature, but reusing the buckets makes sense here. Thanks for clearing that up for me!

Reply:
Np! The naming I will admit is confusing here. I could do something like `bucketConstructId` to make it more clear.